src/transformations/simplify.jl

```diff
@@ -138,42 +139,37 @@ Simplifies geometries by removing points less than
 
 $SIMPLIFY_ALG_KEYWORDS
 - `tol`: the minimum distance between points.
+
+Note: user input `tol` is squared to avoid uneccesary computation in algorithm.
 """
-struct RadialDistance <: SimplifyAlg 
-    number::Union{Int64,Nothing}
-    ratio::Union{Float64,Nothing}
-    tol::Union{Float64,Nothing}
-end
-function RadialDistance(; number=nothing, ratio=nothing, tol=nothing)
-    checkargs(number, ratio, tol)
-    return RadialDistance(number, ratio, tol)
+@kwdef struct RadialDistance <: SimplifyAlg 
+    number::Union{Int64,Nothing} = nothing
+    ratio::Union{Float64,Nothing} = nothing
+    tol::Union{Float64,Nothing} = nothing
+
+    function RadialDistance(number, ratio, tol)
+        _checkargs(number, ratio, tol)
+        # square tolerance for reduced computation
+        tol = isnothing(tol) ? tol : tol^2
+        new(number, ratio, tol)
+    end
 end
 
-settol(alg::RadialDistance, tol) = RadialDistance(alg.number, alg.ratio, tol)
-
 function _simplify(alg::RadialDistance, points::Vector)
     previous = first(points)
     distances = Array{Float64}(undef, length(points))
     for i in eachindex(points)
         point = points[i]
-        distances[i] = _squared_dist(point, previous)
+        distances[i] = _squared_euclid_distance(Float64, point, previous)
         previous = point
     end
     ## Never remove the end points
     distances[begin] = distances[end] = Inf
-    ## This avoids taking the square root of each distance above
```

Comment:
The problem with this for me is that if you make an object

`alg = RadialDistance(; tol = 1.5)` and you have two polygons `poly1` and `poly2`, I would expect to be able to do the following:

`simplify(alg, poly1)`
`simplify(alg, poly2)`

But in this case, the tolerance would be squared twice, right? I would rather have a converter to take a `DouglasPeucker` or a `VisvalingamWhyatt` algorithm to a `RadialDistance` algorithm and it know to either square it or halve if or whatever and then pass that to use the pre-filtering.

Reply:
Actually, it might make sense for them to just provide a pre-filtering algorithm if they so desire with an optional `prefilter_alg` argument. This would increase the flexibility and also would be more consistent than using `tol` since the `tol` for `VisvalingamWhyatt` is an area while the `tol` for both of the other two is a distance which would cause problems if you try to prefilter `VisvalingamWhyatt` using `RadialDistance` with the same `alg` object. This would then allow a different tolerance/number/ratio for the pre-filtering than the actual simplification algorithm and would let people hook into the architecture easier. Thoughts?

Reply:
Yeah thats a better idea.